### PR TITLE
Add Buildah buildstep plugin

### PR DESCRIPTION
**Pull request by rcerven**
Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/build_buildah.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            # include in the failure summary.
+            err = output[-1] if output else "<buildah had bad exit code but no output>"
+            return BuildResult(
+                logs=output,
+                fail_reason="image build failed (rc={}): {}".format(ib_process.returncode, err),
+            )
+
+        image_id = builder.get_built_image_info()['Id']
+        if ':' not in image_id:
+            # Older versions of the daemon do not include the prefix
```
Comment (athos-ribeiro) [inline atomic_reactor/plugins/build_buildah.py]: Does this block still make sense here? This seems to be a docker daemon specific issue, but this image_id would have to be fetched by podman, right?

File: atomic_reactor/plugins/build_buildah.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            image_id = 'sha256:{}'.format(image_id)
+
+        # since we need no squash, export the image for local operations like squash would have
+        self.log.info("fetching image %s from docker", image)
```
Comment (athos-ribeiro) [inline atomic_reactor/plugins/build_buildah.py]: Same idea here, wouldn't we need to push the built image to the docker daemon storage first in orther to fetch it here?

File: atomic_reactor/plugins/build_buildah.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/build_buildah.py]: seems like docker daemon specific, but I have no way to verify that

File: atomic_reactor/plugins/build_buildah.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/build_buildah.py]: mmm? keep in mind that this plugin will use PodmanTasker which isn't implemented yet

**Comment by athos-ribeiro** [comment]
LGTM, just a couple inline questions.

Should we start documenting this feature right away? I mean, should this already include a PR for osbs-docs?

Also, the 1st commit does not have a signed-off-by line. I wouldn't oppose to waiving that one for this PR though.

**Comment by twaugh** [comment]
I would leave out the documentation for the time being, until we have a plugin that can actually work.

We should have Signed-Off-By though.

**Comment by athos-ribeiro** [comment]
Waiving third party signed off by entry

**Comment by rcerven** [comment]
rn updated: - new Buildah buildstep plugin

**Pull request merged**